WordPress-iOS-Shared/Core/WPTableViewSectionHeaderFooterView.m

```diff
@@ -0,0 +1,227 @@
+#import "WPTableViewSectionHeaderFooterView.h"
+#import "WPTableViewCell.h"
+#import "WPStyleGuide.h"
+#import "NSString+Util.h"
+
+
+
+@interface WPTableViewSectionHeaderFooterView ()
+@property (nonatomic, strong) UILabel *titleLabel;
+@end
+
+
+@implementation WPTableViewSectionHeaderFooterView
+
+- (instancetype)initWithStyle:(WPTableViewSectionStyle)style
+{
+    NSParameterAssert(style == WPTableViewSectionStyleHeader || style == WPTableViewSectionStyleFooter);
+    
+    self = [super init];
+    if (self) {
+        // Defaults
+        BOOL isHeader                   = style == WPTableViewSectionStyleHeader;
+        UIEdgeInsets titleInsets        = isHeader ? [[self class] headerTitleInsets] : [[self class] footerTitleInsets];
+        UIColor *titleColor             = isHeader ? [[self class] headerTitleColor]  : [[self class] footerTitleColor];
+        UIFont *titleFont               = isHeader ? [[self class] headerTitleFont]   : [[self class] footerTitleFont];
+        
+        // Title Label
+        UILabel *titleLabel             = [[UILabel alloc] init];
+        titleLabel.textAlignment        = NSTextAlignmentLeft;
+        titleLabel.numberOfLines        = 0;
+        titleLabel.lineBreakMode        = NSLineBreakByWordWrapping;
+        titleLabel.font                 = titleFont;
+        titleLabel.textColor            = titleColor;
+        titleLabel.backgroundColor      = [UIColor clearColor];
+        titleLabel.shadowOffset         = CGSizeZero;
+        [self addSubview:titleLabel];
+
+        // Background
+        UIView *backgroundView          = [UIView new];
+        backgroundView.backgroundColor  = [UIColor clearColor];
+        
+        // Initialize Prperties
+        self.uppercase                  = isHeader;
```

Comment:
You should use the ivars instead of the properties methods on the initializer, but I assume you want the side effects of the properties changes to apply?